### ISSUE TITLE
Dependency genelet/determined containing BUSL-1.1 licensed code

### ISSUE THREAD

**Issue by johanneskastl** [gruntwork-io/terragrunt#3078]
I have been packaging terragrunt for openSUSE for a long time. Suddenly the checks reported that one of the dependencies contains BUSL-1-1 licensed code:

```
    BUSL-1.1: 17 files
        vendor/github.com/genelet/determined/internal/ipaddr/doc.go
        vendor/github.com/genelet/determined/internal/lang/funcs/cidr.processed.go
        vendor/github.com/genelet/determined/internal/lang/funcs/collection.processed.go
        vendor/github.com/genelet/determined/internal/lang/funcs/conversion.processed.go
        vendor/github.com/genelet/determined/internal/lang/funcs/crypto.processed.go
        vendor/github.com/genelet/determined/internal/lang/funcs/datetime.go
        vendor/github.com/genelet/determined/internal/lang/funcs/descriptions.processed.go
        vendor/github.com/genelet/determined/internal/lang/funcs/encoding.processed.go
        vendor/github.com/genelet/determined/internal/lang/funcs/filesystem.processed.go
        vendor/github.com/genelet/determined/internal/lang/funcs/number.go
        vendor/github.com/genelet/determined/internal/lang/funcs/redact.go
        vendor/github.com/genelet/determined/internal/lang/funcs/refinements.go
        vendor/github.com/genelet/determined/internal/lang/funcs/sensitive.go
        vendor/github.com/genelet/determined/internal/lang/funcs/string.go
        vendor/github.com/genelet/determined/internal/lang/marks/marks.go
        vendor/github.com/genelet/determined/internal/lang/types/type_type.go
        vendor/github.com/genelet/determined/internal/lang/types/types.go
```

This comes from this line in the go.mod file:

https://github.com/gruntwork-io/terragrunt/blob/945145bfe5aabed02732e60d3065ad230dd8216f/go.mod#L59

IANAL, but apparently this means that openSUSE can no longer safely package terragrunt. Not sure what that means for terragrunt itself, as BUSL-1-1 and MIT do not play well with each other. So this issue is both for making you aware of this issue and asking for clarification.

Thanks in advance,
Johannes

**Comment by levkohimins** [comment]
@johanneskastl, thank you. We don't really need the code under this license, but unfortunately the package we import for some specific functionality uses it. Since that package is not very popular, and our needs for it are very modest, I think it’s worth writing our own simple code. I will do it.

**Comment by johanneskastl** [comment]
Thanks @levkohimins 
that would be very nice!

**Comment by levkohimins** [comment]
Resolved in [v0.57.6](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.57.6) release.

**Comment by johanneskastl** [comment]
Thanks a lot, that was quick!
